tests/api_tests/users/test_views.py

```diff
@@ -203,6 +206,241 @@ def test_get_projects_logged_in_as_different_user(self):
+        assert_not_equal(res.json['data']['linkedIn'], self.newLinkedIn)
+        assert_not_equal(res.json['data']['impactStory'], self.newImpactStory)
+        assert_not_equal(res.json['data']['orcid'], self.newOrcid)
+        assert_not_equal(res.json['data']['researcherId'], self.newResearcherId)
+
+
+
+
+    def test_put_wrong_user(self):
+        # User tries to update someone else's user information via put
+        put = self.app.put_json(self.user_one_url, {
+            'id': self.user_one._id,
+            'fullname': self.new_fullname,
+            'given_name': self.new_given_name,
+            'family_name': self.new_family_name,
+            'suffix': self.new_suffix,
+            'github': self.newGithub,
+            'personal_website': self.newPersonal_website,
+            'twitter': self.newTwitter,
+            'linkedIn': self.newLinkedIn,
+            'impactStory': self.newImpactStory,
+            'orcid': self.newOrcid,
+            'researcherId': self.newResearcherId,
+        }, auth=self.auth_two, expect_errors=True)
+        # This is 403 instead of 401 because basic authentication is only for unit tests and, in order to keep from
+        # presenting a basic authentication dialog box in the front end. We may change this as we understand CAS
+        # a little better
+        assert_equal(put.status_code, 403)
+        res = self.app.get(self.user_one_url)
+        assert_not_equal(res.json['data']['fullname'], self.new_fullname)
+        assert_not_equal(res.json['data']['given_name'], self.new_given_name)
+        assert_not_equal(res.json['data']['family_name'], self.new_family_name)
+        assert_not_equal(res.json['data']['suffix'], self.new_suffix)
+        assert_not_equal(res.json['data']['github'], self.newGithub)
+        assert_not_equal(res.json['data']['personal_website'], self.newPersonal_website)
+        assert_not_equal(res.json['data']['twitter'], self.newTwitter)
+        assert_not_equal(res.json['data']['linkedIn'], self.newLinkedIn)
+        assert_not_equal(res.json['data']['impactStory'], self.newImpactStory)
+        assert_not_equal(res.json['data']['orcid'], self.newOrcid)
+        assert_not_equal(res.json['data']['researcherId'], self.newResearcherId)
+
+
+    def test_patch_wrong_user(self):
+        # User tries to update someone else's user information via patch
+        patch = self.app.patch_json(self.user_one_url, {
+            'fullname': self.new_fullname,
+        }, auth=self.auth_two, expect_errors=True)
+        # This is 403 instead of 401 because basic authentication is only for unit tests and, in order to keep from
+        # presenting a basic authentication dialog box in the front end. We may change this as we understand CAS
+        # a little better
+        assert_equal(patch.status_code, 403)
+        res = self.app.get(self.user_one_url)
+        assert_not_equal(res.json['data']['fullname'], self.new_fullname)
```

Comment:
It would be better to assert equal to the old fullname to verify that it wasn't changed, in case somehow the information were deleted or in some other way changed to something that is not the new name but also not the old name. Same for the other not_equal tests.